platform/ios/MGLAccountManager.m

```diff
@@ -0,0 +1,50 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLAccountManager.h"
+#import "NSProcessInfo+MGLAdditions.h"
+#import "MGLMapboxEvents.h"
+
+@interface MGLAccountManager()
+
+@property (atomic) NSString *accessToken;
+
+@end
+
+
+@implementation MGLAccountManager
+
+static MGLAccountManager *_sharedManager;
+
+// Can be called from any thread. Called implicitly from any
+// public class convenience methods.
+//
++ (instancetype) sharedInstanceWithAccessToken:(NSString *)token {
```

Comment:
If I understand correctly using the instance methods would then have the initial setup in the AppDelegate be something like:

```
[[MGLAcountManager sharedManager] setAccessToken:@"FOO"]

[MGLAcountManager sharedInstanceWithAccessToken:@"FOO"]

```

How about renaming it to `managerWithInititalAccessTokenValue` or a derivative?

FWIW, I'm not terribly fond of the C style option so I'd like to get the first option working.